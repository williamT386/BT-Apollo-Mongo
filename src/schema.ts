import { gql } from 'apollo-server';
import user from './user/user.schema.js';

const sharedSchema = gql`
    enum UserRole {
        USER
        ADMIN
    }
`;

export default [sharedSchema, user];
