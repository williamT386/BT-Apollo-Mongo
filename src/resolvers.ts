import user from './user/user.resolver.js';

export default [user];
