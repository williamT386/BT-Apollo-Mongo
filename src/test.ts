import mongodb from 'mongodb'

var client = mongodb.MongoClient;
var url = "mongodb://localhost:27017/";

const client.connect(url, function (err, client) {
    
    var db = client.db("bt");
    var collection = db.collection("user");
    
    var query = {};
    
    var cursor = collection.find(query);
    
    cursor.forEach(
        function(doc) {
            console.log(doc);
        }, 
        function(err) {
            client.close();
        }
    );

    
});