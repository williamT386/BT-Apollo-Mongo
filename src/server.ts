import { ApolloServer } from 'apollo-server';
import { mergeSchemas } from 'graphql-tools';
import schemas from './schema.js';
import resolvers from './resolvers.js';
import { getConfig } from './util';

const schema = mergeSchemas({
    schemas,
    resolvers
});

export const serverConfig = {
    schema,
    context: ({ req }) => ({
        // token: req.headers['authorization'],
        organizationId: req.headers['organization-id']
    })
};

export const instantiateServer = (customConfig = {}) =>
    new ApolloServer({
        ...serverConfig,
        ...customConfig
    });

export async function startServer(server = instantiateServer()) {
    try {
        const port = 4001
        const { url } = await server.listen(port && { port });
        console.log(`ðŸš€  Apollo Server ready at ${url}.`);
    } catch (error) {
        console.error('Could not start server.', error);
    }
}
