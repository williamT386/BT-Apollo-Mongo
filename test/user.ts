import { createTestClient } from 'apollo-server-testing';
import { instantiateServer } from '../src/server';
import {
    createDatabaseConnection,
    closeDatabaseConnection,
    dropDatabase
} from '../src/database';
import {
    USERS,
    USER_BY_ID,
    CREATE_USER,
    UPDATE_USER
} from './fixture/queries.user';

const TEST_DB_CONNECTION_STRING = 'mongodb://localhost:27017/template-db-test';

let query = null;
const token = '5e7abd771f06a07014d872c0';
const organizationId = '5e7abc6792323e650e6c76e2';

describe('User:', function() {
    beforeAll(async function() {
        await createDatabaseConnection(TEST_DB_CONNECTION_STRING);
        const server = instantiateServer({
            context: () => ({
                token,
                organizationId
            })
        });
        ({ query } = createTestClient(server));
    });
    afterAll(async function() {
        await dropDatabase();
        await closeDatabaseConnection();
    });

    test('Create a user', async function() {
        const result = await query({
            query: CREATE_USER,
            variables: {
                role: 'ADMIN',
                pinCode: '0519'
            }
        });
        expect(result).toMatchSnapshot();
    });

    test('Update a user', async function() {
        const result = await query({
            query: UPDATE_USER,
            variables: {
                id: '1', // id from previously created user
                role: 'USER',
                pinCode: '7068'
            }
        });
        expect(result).toMatchSnapshot();
    });

    test('Fetch user by identifier', async function() {
        const result = await query({
            query: USER_BY_ID,
            variables: {
                id: '1' // id from previously created user
            }
        });

        expect(result).toMatchSnapshot();
    });

    test('Fetch users', async function() {
        const result = await query({
            query: USERS
        });

        expect(result).toMatchSnapshot();
    });
});
