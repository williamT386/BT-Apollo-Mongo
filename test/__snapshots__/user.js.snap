// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User: Create a user 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "id": "1",
      "organizationId": "5e7abc6792323e650e6c76e2",
      "pinCode": "0519",
      "role": "ADMIN",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`User: Fetch user by identifier 1`] = `
Object {
  "data": Object {
    "userById": Object {
      "id": "1",
      "organizationId": "5e7abc6792323e650e6c76e2",
      "pinCode": "7068",
      "role": "USER",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`User: Fetch users 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "id": "1",
        "organizationId": "5e7abc6792323e650e6c76e2",
        "pinCode": "7068",
        "role": "USER",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`User: Update a user 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "id": "1",
      "organizationId": "5e7abc6792323e650e6c76e2",
      "pinCode": "7068",
      "role": "USER",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
